components/core/Routes/Download.py

```diff
@@ -0,0 +1,168 @@
+import sys, os
+dir_path = os.path.abspath(__file__)[:-19]
+sys.path.append(dir_path)
+
+from flask import Flask
+from flask.ext.cors import CORS
+from flask import send_file, send_from_directory
+from flask import request, jsonify, abort, Response, g
+from flask_socketio import SocketIO, join_room
+from Auth import *
+from Models import *
+from DownloadManager import *
+import json, urllib.request, urllib.error, urllib.parse, _thread
+from multiprocessing import Process
+from DownloadDaemon import starter
+from EMail import send_mail
+from gevent import monkey
+
+@server.route('/download/start')
+def start():
+    try:
+        token = request.headers['key']
+        if str(token)!=server.config['SECRET_KEY']:
+            return "{'error':'not authorized'}", 403
+        global p
+        p = Process(target=starter, args=(socketio,))
+        p.start()
+        return '{"status":"' + str(p.pid) + '"}'
+    except Exception as e:
+            return '{"error":"' + e.message + '"}',400
+
+@socketio.on('join', namespace='/progress')
+def on_join(data):
+    room = data['room']
+    if room != '':
+        join_room(room)
+
+@server.route('/download/kill')
+def kill():
+    try:
+        token = request.headers['key']
+        if str(token)!=server.config['SECRET_KEY']:
+            return "{'error':'not authorized'}", 403
+        if p is not None:
+            p.terminate()
+            p.join()
+            jsonreq = json.dumps({'jsonrpc':'2.0', 'id':'qwer', 'method':'aria2.pauseAll'})
+            jsonreq = jsonreq.encode('ascii')
+            c = urllib.request.urlopen('http://localhost:6800/jsonrpc', jsonreq)
```

Comment:
localhost might fail when deploying to production.

Reply:
I did not make this :worried: 
I've just separated all the download related functions from the earlier named Rest.py into this file.

Reply:
@rehrumesh, please take a look [here](https://github.com/scorelab/Bassa/issues/462), I was referring the same 😊